modules/apps/commerce/commerce-payment-method-paypal/src/main/java/com/liferay/commerce/payment/method/paypal/internal/PayPalCommercePaymentMethod.java

```diff
@@ -847,12 +847,16 @@ private Agreement _getAgreement(
 
 		agreement.setPayer(payer);
 
-		ShippingAddress shippingAddress = _getShippingAddress(
-			commerceOrder.getShippingAddress());
+		CommerceAddress commerceAddress = commerceOrder.getShippingAddress();
```

Comment:
At least 1 address should be mandatory for the payment provider. It is something agreed long ago that also into Commerce there must be at least 1 address during the checkout to properly fulfil all data needed, even for non shippable items.
That said even if this code is preventing the crash/exception, we will probably end up with the payment provider aborting the payment because of missing data.
If we want to use a checkout process without asking for an address when there's no shippable item in the cart, we should have at least a default shipping address linked to the buyer account.

Reply:
@luca-pellizzon about this? I've changed the previous commit to meet the requirements you said, now we would have 2 way to determine which address to use, one is the default from account and the other is to use the billing address set in the order which I believe is mandatory.

Reply:
I guess this is it. I think there's nothing we can do more.